docs/elasticsearch/search-template.md

```diff
@@ -0,0 +1,416 @@
+
+Search templates are expressed using the Mustache language. For a list of all syntax options, see the [Mustache manual](http://mustache.github.io/mustache.5.html).
+{: .note }
+
+---
+
+#### Table of contents
+1. TOC
+{:toc}
+
+## Create search templates
+
+A search template has two components: the query and the parameters. Parameters are user inputted values that get placed into variables. Variables are represented with double braces in Mustache notation. When encountering a variable like `{% raw %}{{var}}{% endraw %}` in the query, Elasticsearch goes to the `params` section and looks for a parameter called `var` and replaces it with the specified value.
+
+You can code your application to ask your user what they want to search for and then plug in that value in the `params` object at run time.
+
+This command defines a search template to find a play by its name. The `{% raw %}{{play_name}}{% endraw %}` in the query is replaced by the value `Henry IV`:
+
+```json
+GET _search/template
+{
+  "source": {
+    "query": {
+      "match": {
+        "play_name": "{% raw %}{{play_name}}{% endraw %}"
+      }
+    }
+  },
+  "params": {
+    "play_name": "Henry IV"
+  }
+}
+```
+
+This template runs the search on your entire cluster.
+To run this search on a specific index, add the index name to the request:
+
+```json
+GET shakespeare/_search/template
+```
+
+You can implement pagination and show a set number of search results at a time using the `from` and `size` parameters:
+
+```json
+GET _search/template
+{
+  "source": {
+    "from": "{% raw %}{{page}}{% endraw %}",
+    "size": "{% raw %}{{size}}{% endraw %}",
+    "query": {
+      "match": {
+        "play_name": "{% raw %}{{play_name}}{% endraw %}"
+      }
+    }
+  },
+  "params": {
+    "play_name": "Henry IV",
+    "page": 0,
+    "size": 10
+  }
+}
+```
+
+To improve the search experience, you can define defaults so that the user doesn’t have to specify every possible parameter. If the parameter is not defined in the `params` section, the default is used.
+
+The syntax for defining the default value for a variable `var` is as follows:
+
+```json
+{% raw %}{{var}}{{^var}}default value{{/var}}{% endraw %}
+```
+
+This command sets the defaults for page number as 0 and number of results per page as 10.
+
+```json
+GET _search/template
+{
+  "source": {
+    "from": "{% raw %}{{page}}{{^page}}0{{/page}}{% endraw %}",
+    "size": "{% raw %}{{size}}{{^size}}10{{/size}}{% endraw %}",
+    "query": {
+      "match": {
+        "play_name": "{% raw %}{{play_name}}{% endraw %}"
+      }
+    }
+  },
+  "params": {
+    "play_name": "Henry IV"
+  }
+}
+```
+
+
+## Save and execute search templates
+
+Once you have the search template working the way you want it to, you can save the source of that template as a script making it reusable for different input parameters.
```

Comment:
After you have the search template working the way you want it to, you can save the source of that template as a script, making it reusable for different input parameters.